README.md

```diff
@@ -330,6 +331,65 @@ end
 
 5. Your app is now able to access the authenticated `ProductsController` from the `HomeController` using session tokens.
 
+### Marking shops as uninstalled on app/uninstalled webhook
+
+To ensure OAuth still works with session tokens, your app must update its shop records in the event a shop uninstalls your app. You can be notified of these events by subscribing to the `app/uninstall` [webhook](https://shopify.dev/docs/admin-api/rest/reference/events/webhook).
+
+#### Setup app/uninstall webhook
+1. Use the `add_webhook` generator from ShopifyApp to setup the `app/uninstalled` webhook for the app.
+
+```shell script
+rails g shopify_app:add_webhook -t app/uninstalled -a {your_app_url}/webhooks/app_uninstalled
+```
+
+This will add the `app/uninstalled` webhook to your config and create the `AppUninstalledJob` job class for you.  
+
+2. Update the `AppUninstalledJob` job class to mark the Shop record as uninstalled.
+
+```ruby
+class AppUninstalledJob < ActiveJob::Base
+  def perform(args)
+    shop = Shop.find_by(shopify_domain: args[:shop_domain])
+
+    mark_shop_as_uninstalled(shop)
+  end
+
+  private
+
+  def mark_shop_as_uninstalled(shop)
+    shop.destroy! if shop
```

Comment:
I wonder if we should make this demonstration of soft-deleting the shop so that any additional data in the shop model isn't lost.

Reply:
+1. That would be very helpful to give app devs who currently rely on data persisting through reinstalls have a clear/reliable path forward.

Reply:
We can't do soft delete. Shopify app doesn't use the soft delete gem and soft delete uses a different table structure.

Reply:
This would involve adding a variation from what shopify_app provides as well. I don't think the app should be deviating from default behaviour.

Reply:
> Shopify app doesn't use the soft delete gem

There's a separate gem for soft deletion? :o

I think what Sam and I were referring to was simply "not permanently removing the shop record upon uninstall".
If this is not recommended, we should call it out in the docs as well since apps relying on persisting related data through reinstalls will be impacted by migrating to jwt.

Reply:
Agreed. I'm reaching out to DTE to see what their recommendation is for this.